diff --git a/python/CMakeLists.txt b/python/CMakeLists.txt
index ac23871ce..d234cec51 100644
--- a/python/CMakeLists.txt
+++ b/python/CMakeLists.txt
@@ -27,24 +27,11 @@ else()
     set(QtWidgets_VERSION ${Qt5Widgets_VERSION})
 endif()
 
-find_package(Python3 3.7 REQUIRED COMPONENTS Interpreter Development)
+find_package(Python3 3.7 REQUIRED COMPONENTS Interpreter Development.Module)
 find_package(Shiboken${PYSIDE_MAJOR_VERSION} REQUIRED)
 find_package(PySide${PYSIDE_MAJOR_VERSION} ${QtWidgets_VERSION} EXACT REQUIRED)
 
-if(NOT ${PROJECT_NAME}_PYTHON_BINDINGS_INSTALL_PREFIX)
-    # cmake-lint: disable=C0103
-    set(${PROJECT_NAME}_PYTHON_BINDINGS_INSTALL_PREFIX
-        ${INSTALL_LIBRARY_DIR}
-        CACHE FILEPATH "Custom path to install python bindings."
-    )
-endif()
-if(NOT WIN32)
-    set(Python3_VERSION_MAJORMINOR "${Python3_VERSION_MAJOR}.${Python3_VERSION_MINOR}")
-    set(BINDINGS_DIR "python${Python3_VERSION_MAJORMINOR}/site-packages/${PYTHON_BINDING_NAMESPACE}")
-else()
-    set(BINDINGS_DIR "site-packages/${PYTHON_BINDING_NAMESPACE}")
-endif()
-set(${PROJECT_NAME}_PYTHON_BINDINGS_INSTALL_PREFIX "${${PROJECT_NAME}_PYTHON_BINDINGS_INSTALL_PREFIX}/${BINDINGS_DIR}")
+set(${PROJECT_NAME}_PYTHON_BINDINGS_INSTALL_PREFIX "${PYTHON_BINDING_NAMESPACE}")
 
 include(KDPySide${PYSIDE_MAJOR_VERSION}ModuleBuild)
 add_subdirectory(PyKDDockWidgets)
diff --git a/python/PyKDDockWidgets/CMakeLists.txt b/python/PyKDDockWidgets/CMakeLists.txt
index 8e9512fd7..606f1e729 100644
--- a/python/PyKDDockWidgets/CMakeLists.txt
+++ b/python/PyKDDockWidgets/CMakeLists.txt
@@ -41,7 +41,7 @@ set(PyKDDockWidgets_typesystem_paths # PySide path, this variable was exposed by
     ${PYSIDE_TYPESYSTEMS}
 )
 # Include flags/path that will be set in 'target_include_directories'
-set(PyKDDockWidgets_target_include_directories ${CMAKE_SOURCE_DIR}/src)
+set(PyKDDockWidgets_target_include_directories ${PROJECT_SOURCE_DIR}/src)
 
 # Libraries that will be necessary to link the target, this will used in the
 # command 'target_link_libraries'
@@ -51,9 +51,9 @@ set(PyKDDockWidgets_target_link_libraries KDAB::kddockwidgets Qt${QT_VERSION_MAJ
 
 # changes on these files should trigger a new generation
 set(PyKDDockWidgets_DEPENDS
-    ${CMAKE_CURRENT_SOURCE_DIR}/kddockwidgets_global.h ${CMAKE_SOURCE_DIR}/src/qtwidgets/views/MainWindow.h
-    ${CMAKE_SOURCE_DIR}/src/qtwidgets/views/DockWidget.h ${CMAKE_SOURCE_DIR}/src/Config.h
-    ${CMAKE_SOURCE_DIR}/src/LayoutSaver.h
+    ${CMAKE_CURRENT_SOURCE_DIR}/kddockwidgets_global.h ${PROJECT_SOURCE_DIR}/src/qtwidgets/views/MainWindow.h
+    ${PROJECT_SOURCE_DIR}/src/qtwidgets/views/DockWidget.h ${PROJECT_SOURCE_DIR}/src/Config.h
+    ${PROJECT_SOURCE_DIR}/src/LayoutSaver.h
 )
 
 create_python_bindings(
diff --git a/python/PyKDDockWidgets/typesystem_kddockwidgets.xml b/python/PyKDDockWidgets/typesystem_kddockwidgets.xml
index 8d2171cde..0ab2cff0d 100644
--- a/python/PyKDDockWidgets/typesystem_kddockwidgets.xml
+++ b/python/PyKDDockWidgets/typesystem_kddockwidgets.xml
@@ -63,7 +63,7 @@
         </object-type>
 
         <namespace-type name="Core" visible="no">
-            <enum-type name="ViewType" />
+            <!-- <enum-type name="ViewType" /> workaround: shiboken 6.10.2 generates duplicate symbols -->
             <object-type name="DockWidgetViewInterface" >
               <include file-name="kddockwidgets/core/views/DockWidgetViewInterface.h" location="global"/>
             </object-type>
